addons/udes_stock/models/stock_move.py

```diff
@@ -118,6 +131,42 @@ def _unreserve_initial_demand(self, new_move):
             {"move_id": new_move.id, "product_uom_qty": 0}
         )
 
+    def propagate_cancellation_to_next_picks(self, old_dest_ids):
+        """
+        Propagate cancelled qtys to move_dest_ids.
+        If a full qty match is found, cancel that move - calling this function for that move
+        If a part qty match is found, decrement the qty from the move, and continue down the chain
+
+        `propagate_cancel` on `stock.rule` could be used, however it is not visible on `Push` rules
+        and does not appear to propagate partial cancellations.
+
+        :param: old_dest_ids: stock.move(x) recordset(s), fallback if no move_dest_ids in self.
+                              Can be emptyset
+        """
+        StockMove = self.env["stock.move"].browse()
+        next_moves_filter = lambda m: m.picking_type_id.u_propagate_cancel == True
+        next_moves = self.move_dest_ids.filtered(next_moves_filter)
+        if not next_moves:
+            # NB: This handles a symptom of Issue #2771
+            next_moves = old_dest_ids
```

Comment:
Q: what is the expected behaviour when we have a picking_type down the chain with `u_propagate_cancel =  False`? Will we still expect that picking_types moves to be cancelled?

Reply:
Yeah, thinking about it I think i have the `.filtered(next_moves_filter)` in the wrong places. I'll review and push an update

Having `u_propagate_cancel` set to True on the pick type being cancelled will cancel the next moves, but if those next moves have `u_propagate_cancel` set to False, the next moves of those will not be cancelled

Reply:
Yeah cool, I just wanted to make sure I understood correctly :smile:

Reply:
I've changed to only do the filter when we have part cancellations, and change `next_moves` inline in the while loop. This is because inside `_action_cancel()` there is a check, so for all other cases it doesn't need filtering as it only calls `propagate_cancellation_to_next_picks()` when those moves picking types have `u_propagate_cancel` set to True